packages/react-instantsearch-core-next/src/InstantSearch.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+import { IndexContext } from './IndexContext';
+import { InstantSearchContext } from './InstantSearchContext';
+import { useInstantSearch } from './useInstantSearch';
+
+import type { UseInstantSearchProps } from './useInstantSearch';
+
+type InstantSearchProps = UseInstantSearchProps & {
+  children?: React.ReactNode;
+};
+
+export function InstantSearch({ children, ...props }: InstantSearchProps) {
+  const search = useInstantSearch(props);
+
+  return (
+    <InstantSearchContext.Provider value={search}>
+      <IndexContext.Provider value={search.mainIndex}>
```

Comment:
this is genius! it could be implemented in Vue InstantSearch too so the inject is no longer needing to fall back to mainIndex, but it's InstantSearch providing it!

Reply:
Yeah let's use the framework!